themes/bootstrap3/js/account_ajax.js

```diff
@@ -0,0 +1,177 @@
+/*global userIsLoggedIn, VuFind */
+VuFind.register('account', function Account() {
+  var LOADING = -1;
+  var MISSING = -2;
+  var _sessionDataKey = 'account-statuses';
+
+  var checkedOutStatus = LOADING;
+  var fineStatus = LOADING;
+  var holdStatus = LOADING;
+
+  var _save = function _save() {
+    sessionStorage.setItem(_sessionDataKey, JSON.stringify({
+      checkedOut: checkedOutStatus,
+      fines: fineStatus,
+      holds: holdStatus
+    }));
+  };
+
+  // Clearing save forces AJAX update next page load
+  var _clearSave = function _clearSave() {
+    sessionStorage.removeItem(_sessionDataKey);
+  };
+
+  var _render = function _render() {
+    function formatNumber(num) {
+      return num.toFixed(2).replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+    }
+    var accountIcon = 'fa fa-user-circle';
+    // CHECKED OUT COUNTS
+    if (checkedOutStatus === MISSING) {
+      $('.myresearch-menu .checkedout-status').addClass('hidden');
+    } else {
+      var html = '';
+      if (checkedOutStatus !== LOADING) {
+        if (checkedOutStatus.ok > 0) {
+          html += '<span class="badge ok">' + checkedOutStatus.ok + '</span>';
+        }
+        if (checkedOutStatus.warn > 0) {
+          html += '<span class="badge warn">' + checkedOutStatus.warn + '</span>';
+          accountIcon = 'fa fa-book text-warning';
+        }
+        if (checkedOutStatus.overdue > 0) {
+          html += '<span class="badge overdue">' + checkedOutStatus.overdue + '</span>';
+          accountIcon = 'fa fa-book text-danger';
+        }
+      }
+      $('.myresearch-menu .checkedout-status').html(html);
+      $('.myresearch-menu .checkedout-status').removeClass('hidden');
+    }
+    // HOLDS
+    if (holdStatus === 'PICKUP') {
+      $('.myresearch-menu .holds-status')
+        .removeClass('hidden fa-spin fa-spinner')
+        .removeClass('fa-clock-o warn')
+        .addClass('fa-bell ok');
+      accountIcon = 'fa fa-bell text-success';
+    } else if (holdStatus === 'INTRANSIT') {
+      $('.myresearch-menu .holds-status')
+        .removeClass('hidden fa-spin fa-spinner')
+        .removeClass('fa-bell ok')
+        .addClass('fa-clock-o warn');
+    } else if (holdStatus !== LOADING) {
+      $('.myresearch-menu .holds-status').addClass('hidden');
+    }
+    // FINES
+    if (fineStatus === MISSING) {
+      $('.myresearch-menu .fines-status').addClass('hidden');
+    } else if (fineStatus === LOADING) {
+      $('.myresearch-menu .fines-status').html(
+        '<i class="fa fa-spin fa-spinner" aria-hidden="true"></i>'
+      );
+    } else {
+      $('.myresearch-menu .fines-status').html(
+        '<span class="badge overdue">$' + formatNumber(fineStatus / 100) + '</span>'
```

Comment:
We can't assume dollars here, nor should we code our own number-formatting function; these details are handled by PHP's localization system, and we should try to be consistent with that. The SafeMoneyFormat view helper does this job; maybe it would make more sense to simply inject that helper into the AJAX handler for fines and return a formatted rather than raw value to the JS.